onnxruntime/core/session/plugin_ep/ep_plugin_provider_interfaces.cc

```diff
@@ -188,6 +218,14 @@ PluginExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_vie
       indexed_sub_graph->nodes.push_back(node_grouping.nodes[0]->GetInternalNode().Index());
       result.push_back(std::make_unique<ComputeCapability>(std::move(indexed_sub_graph)));
     } else if (node_grouping.kind == OrtEpGraphSupportInfo::NodeGroupingKind::kFusedNode) {
+      if (node_grouping.nodes.empty()) {
+        // The EpGraphSupportInfo_AddNodesToFuse() C API should already return an error if the EP tries to provide
+        // an empty array of nodes from OrtEp::GetCapability(). However, we check here too just in case this changes.
+        LOGS(logger, WARNING) << "OrtEp::GetCapability() for " << Type() << " set an empty array of nodes "
```

Comment:
nit: Should this check and warning also apply to the `NodeGroupingKind::kSingleAssignedNode` case above?

Reply:
Added. Thanks.